src/core/Akka/Actor/ReceiveActorHandlers.cs

```diff
@@ -0,0 +1,200 @@
+// -----------------------------------------------------------------------
+//  <copyright file="ReceiveActorHandlers.cs" company="Akka.NET Project">
+//      Copyright (C) 2009-2025 Lightbend Inc. <http://www.lightbend.com>
+//      Copyright (C) 2013-2025 .NET Foundation <https://github.com/akkadotnet/akka.net>
+//  </copyright>
+// -----------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+
+namespace Akka.Actor;
+#nullable enable
+internal sealed class ReceiveActorHandlers
+{
+    private bool _hadObjectHandlerWithNoPredicate;
+
+    public ReceiveActorHandlers()
+    {
+        TypedHandlers = new List<ITypeHandler>();
+        HandleAny = null;
+    }
+
+    private List<ITypeHandler> TypedHandlers { get; }
```

Comment:
Moved from a `Dictionary` to a `List` in order to guarantee that declared `Receive<T>` order matches our handling order, which is an important behavioral requirement